### ISSUE TITLE
Unexpected Behavior in Unknown Environment Initialization with Pretrained Interactive Model

### ISSUE THREAD

**Issue by FBI-openup** [facebookresearch/nwm#7]
Dear NWM authors,
First of all, thank you for your amazing work on NWM. The framework is elegant and effective, and I truly appreciate the clarity of your code and documentation.

I have been using the interactive model with your pretrained weights, and I was able to reproduce the impressive results in known environments, as described in the paper. Specifically, I used a frame captured from the SCAND dataset, which I named frame, and observed a strong match with the behaviors and performance shown in your examples.

However, I encountered an issue while testing the model's behavior in unknown environments. I modified the choose starting image block to include the following custom image URLs:
```python
# List of image links
image_links = [
    'https://raw.githubusercontent.com/amirbar/amirbar.github.io/refs/heads/master/images/recon.png',
    'https://raw.githubusercontent.com/amirbar/amirbar.github.io/refs/heads/master/images/scand.png',
    'https://raw.githubusercontent.com/amirbar/amirbar.github.io/refs/heads/master/images/sacson.png',
    'https://raw.githubusercontent.com/amirbar/amirbar.github.io/refs/heads/master/images/tartan.png',
    
     #add customized images 
    #known images different from official website
    #'https://raw.githubusercontent.com/FBI-openup/nwm/main/warehouse%20Images/frame.png',
    # unknown warehouse (collapse)
    'https://raw.githubusercontent.com/FBI-openup/nwm/main/warehouse%20Images/realistic-warehouse-photo1.png',
    'https://raw.githubusercontent.com/FBI-openup/nwm/main/warehouse%20Images/warehouse2.png',
    'https://raw.githubusercontent.com/FBI-openup/nwm/main/warehouse%20Images/obstracle+warehouse3.jpg',
    #'https://raw.githubusercontent.com/FBI-openup/nwm/main/warehouse%20Images/Resized.jpg',
    #'https://raw.githubusercontent.com/FBI-openup/nwm/main/warehouse%20Images/realwarehouse.png',
    #'https://raw.githubusercontent.com/FBI-openup/nwm/main/warehouse%20Images/Low.png',
    #unknown living room environment
    #'https://raw.githubusercontent.com/FBI-openup/nwm/main/warehouse%20Images/livingroom.jpg',
    # unknown environment from official website
    #'https://raw.githubusercontent.com/FBI-openup/nwm/main/warehouse%20Images/chateau.png',
    #unknown From roboflow
    #'https://raw.githubusercontent.com/FBI-openup/nwm/main/warehouse%20Images/cornel.png'
]
```
**In all these cases (especially those intended to represent unknown environments), the model appears to collapse immediately at step 1. Then, from about step 7–8 onwards, it stabilizes into a known environment, as if it quickly disregards the unknown starting image.**

For Instance , I  tested using the castle image from amirbar.net website ,which you labeled as the first example of an unknown environment, but I could not reproduce the same results shown there. **Instead, the model again collapsed at the first step and reverted to a known-like setting.**

Is this behavior expected when using the pretrained interactive model with custom images? Or is there something I might have misunderstood or misconfigured?
if there is any special preprocessing or technique applied to the starting image(resizing, normalization, or a specific resolution, format, or camera viewpoint for the model to properly recognize and handle them as "unknown" scenes).
Any guidance or clarification would be greatly appreciated.

Best regards,
Victor ZHANG

**Comment by FBI-openup** [comment]
Also, I noticed from the paper and dataset usage that NWM includes training on RECON, which contains many sequences captured with a fisheye lens, where the image often **has a circular mask or vignetting effect**. How was this handled during training or data preprocessing? And if I use a starting image that includes a similar circular occlusion (e.g., black corners, lens circle), would the model still behave robustly on your side ?

**Comment by amirbar** [comment]
Hi Victor,

Thanks for you interest in our work. Performance in unknown environment is not amazing, but I think it's interesting to see that the model can learn some visual priors that can potentially carry over, and it would probably become better with more data.

One trick I found to be useful for unknown is environment is to use ground truth trajectories from the training data, then apply them on the new image. Using the jupyter notebook might be a bit limiting as the commands are disentangled, and might already bit a "out of domain.  

See below some examples from the videos you shared.

https://github.com/user-attachments/assets/600318ee-75fd-49fe-85e9-717d4d4d8367
https://github.com/user-attachments/assets/eb45f07d-b406-4e30-81f9-92ab06c8c1f0
https://github.com/user-attachments/assets/ed880ff6-b336-4f1b-a3d4-5cbd8e58eada
https://github.com/user-attachments/assets/c318e989-3c20-49e5-97b8-88fba2e2bf00
https://github.com/user-attachments/assets/59b91231-0c58-4ce0-b382-724cc5d81b1c
https://github.com/user-attachments/assets/ab9ff92c-29ad-46aa-abb9-bd6dbb7a2ec8
https://github.com/user-attachments/assets/9a9d6cf0-d214-44cf-b62a-13e4a4021683
https://github.com/user-attachments/assets/f825cc06-3562-481a-b6e9-28196ae7c09b
https://github.com/user-attachments/assets/b6c73f06-cb3e-4c98-8655-1bbd213027c4

**Comment by FBI-openup** [comment]
Thank you for your reply,

Thank you very much for your detailed feedback and for sharing the illustrative video examples 
Your suggestion to re-apply ground truth trajectories from training scenes onto novel images sounds quite interesting. I’d be very interested in reproducing this method. If possible, could you share the ground truth trajectories you mentioned?
Additionally, I was curious about your note regarding the .ipynb interface being potentially “out of domain.” Could you kindly elaborate on why using Jupyter notebooks might introduce out-of-distribution behavior? Is it due to the middle output produced and re -input so we lose accuracy?
Lastly, due to limited training resources at my institution, I wanted to ask:
Are there any smaller variants of the NWM model available (**e.g., CDiT-Small /Medium/large**)? A lightweight version would be very helpful for experimentation and prototyping.
Thanks again for your support

**Comment by amirbar** [comment]
> Your suggestion to re-apply ground truth trajectories from training scenes onto novel images sounds quite interesting. I’d be very interested in reproducing this method. If possible, could you share the ground truth trajectories you mentioned?
I just randomly sampled ~100 trajectories, each one of length 64 time steps, attached it here.
[actions.txt](https://github.com/user-attachments/files/20392564/actions.txt)

> Additionally, I was confused about your note regarding the .ipynb interface being potentially “out of domain.” Could you elaborate on why using Jupyter might be a bit OOD? Is it due to the Intermediate output produced and re -input so we lose accuracy?

The actions I defined in the jupyter notebook are completely disentangled (you either rotate or move), whereas in the real data that doesn't usually happen (you usually have some motion in all axes...)

> Also ,have your team tried any strategies for optimizing or simplifying the NWM architecture but found them (in)effective?
Not much, I'm pretty sure there could be some avenues for optimization as well :)

> Lastly, due to limited training resources at my institution, can i ask for any smaller variants of the NWM model available (e.g., CDiT-Small /Medium/large and their paras)? A lightweight version would be very helpful！

yes, I can include them. I have to train the models again on face blurred data to release, so it would take me few days to get to it

**Comment by amirbar** [comment]
Hi again @FBI-openup, I've added new CDiT-S/B/L model checkpoints.
